source/FreeRTOS_IP.c

```diff
@@ -667,38 +667,50 @@ void vIPNetworkUpCalls( struct xNetworkEndPoint * pxEndPoint )
  */
 static void prvHandleEthernetPacket( NetworkBufferDescriptor_t * pxBuffer )
 {
-    #if ( ipconfigUSE_LINKED_RX_MESSAGES == 0 )
-    {
-        /* When ipconfigUSE_LINKED_RX_MESSAGES is set to 0 then only one
-         * buffer will be sent at a time.  This is the default way for +TCP to pass
-         * messages from the MAC to the TCP/IP stack. */
-        prvProcessEthernetPacket( pxBuffer );
-    }
-    #else /* ipconfigUSE_LINKED_RX_MESSAGES */
-    {
-        NetworkBufferDescriptor_t * pxNextBuffer;
-
-        /* An optimisation that is useful when there is high network traffic.
-         * Instead of passing received packets into the IP task one at a time the
-         * network interface can chain received packets together and pass them into
-         * the IP task in one go.  The packets are chained using the pxNextBuffer
-         * member.  The loop below walks through the chain processing each packet
-         * in the chain in turn. */
-
-        /* While there is another packet in the chain. */
-        while( pxBuffer != NULL )
+    /* When ipconfigUSE_LINKED_RX_MESSAGES is disabled, the network driver will
+     * send only one buffer at a time.  This is the default way for +TCP to pass
+     * messages from the MAC to the TCP/IP stack.
+     * Enabling ipconfigUSE_LINKED_RX_MESSAGES is useful in high network traffic scenarios.
+     * Instead of passing received packets into the IP task one at a time the
+     * network interface can chain received packets together and pass them into
+     * the IP task in one go.  The packets are chained using the pxNextBuffer
+     * member.  The loop below walks through the chain processing each packet
+     * in the chain in turn. */
+
+    /* The while() below is needed when ipconfigUSE_LINKED_RX_MESSAGES is enabled.
+     * The while() below acts like if() when ipconfigUSE_LINKED_RX_MESSAGES is disabled. */
+    while( pxBuffer != NULL )
+    {
+        /* The network interface should not be sending us network buffers
+         * that are not associated with valid interface and end-point. */
+        if( ( pxBuffer->pxInterface == NULL ) || ( pxBuffer->pxEndPoint == NULL ) )
```

Comment:
How about we do these checks in the `prvProcessEthernetPacket` where the `configASSERT`s were added initially.